src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs

```diff
@@ -59,6 +59,23 @@ protected override bool ProcessMetrics(Batch<Metric> metrics, int timeoutMillise
             return this.exporter.Export(metrics) == ExportResult.Success;
         }
 
+        /// <inheritdoc />
+        protected override bool OnCollect(int timeoutMilliseconds)
+        {
+            if (this.supportedExportModes.HasFlag(ExportModes.Push))
+            {
+                return base.OnCollect(timeoutMilliseconds);
+            }
+
+            if (this.supportedExportModes.HasFlag(ExportModes.Pull) && PullMetricScope.IsPullAllowed)
+            {
+                return base.OnCollect(timeoutMilliseconds);
+            }
+
+            // TODO: add some error log
+            return false;
```

Comment:
Would it be more readable for you?

```suggestion
            bool isSupported = this.supportedExportModes.HasFlag(ExportModes.Push);
            isSupported = isSupported || (this.supportedExportModes.HasFlag(ExportModes.Pull) && PullMetricScope.IsPullAllowed);

            if (!isSupported)
            {
                // TODO: add some error log
                return false;
            }

            return base.OnCollect(timeoutMilliseconds);
```

I just prefer to have the "error-handing" indented rather than the "positive" scenario.

Feel free to resolve the comment if you find that it may be less readable or you simply disagree.

Reply:
Don't have strong preference here. I personally don't find big difference, and I slightly prefer the waterfall style since it's easier to comment / debug / set breakpoint.

Let's see if other folks have strong preference.

Reply:
I am OK to resolve the comment (and keep the code as it is) to not waste others' time. As you mentioned, it does not make any big difference  😉